com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ComputeBuilder.java

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.dynamo.bob.pipeline;
+
+import java.io.IOException;
+
+import com.google.protobuf.TextFormat;
+
+import com.dynamo.bob.Bob;
+import com.dynamo.bob.Builder;
+import com.dynamo.bob.BuilderParams;
+import com.dynamo.bob.CompileExceptionError;
+import com.dynamo.bob.Task;
+import com.dynamo.bob.Task.TaskBuilder;
+import com.dynamo.bob.fs.IResource;
+import com.dynamo.bob.ProtoParams;
+import com.dynamo.bob.util.MurmurHash;
+import com.dynamo.render.proto.Compute.ComputeDesc;
+import com.dynamo.render.proto.Material.MaterialDesc;
+
+// For tests
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.Reader;
+import java.io.OutputStream;
+
+@ProtoParams(srcClass = ComputeDesc.class, messageClass = ComputeDesc.class)
+@BuilderParams(name = "Compute", inExts = {".compute"}, outExt = ".computec")
+public class ComputeBuilder extends Builder<Void>  {
+    @Override
+    public Task<Void> create(IResource input) throws IOException, CompileExceptionError {
+        TaskBuilder<Void> task = Task.<Void> newBuilder(this)
+                .setName(params.name())
+                .addInput(input)
+                .addOutput(input.changeExt(params.outExt()));
+
+        ComputeDesc.Builder computeBuilder = ComputeDesc.newBuilder();
+        ProtoUtil.merge(input, computeBuilder);
+
+        IResource computeProgramresource = this.project.getResource(computeBuilder.getComputeProgram()).changeExt(".cpc");
+        task.addInput(computeProgramresource);
```

Comment:
Do we want "#include" support from the start?

Reply:
All shaderbuilders use the include preprocessor so it's already in place. But we should add tests for it for compute as well, I think we only do so for the other shader types